### PR TITLE
Simplification of workflow scripts

### PR DESCRIPTION
**Pull request by Mingun**
Current solution when all scripts run is one step seems not good for me. Running them in different steps gives help from GitHub Actions, which groups output of each step, so for contributors it is easy to see on which step they wrong.

Further, part of scripts just do nothing (except that they checking that they itself running):
- bechmarks (`npm run bench`), because now results is not stored, so no reason to waste time for it
- creating a bundle (`npm run rollup`)
- minification (`npm run terser`)
- appending copyright information to the minified files (`npm run headers`)
- copying build files into target locations (`npm run deploy`)

Moreover, some scripts currently doing the wrong things, in my opinion. They a changing the repository during build which only should be done by the scripts that prepare deploy. Anyway, such scripts should never be called automatically when their result is stored in the repository. It is responsibility of a developer to run them, run tests and commit the result.

For that reasons I've removed all steps that change the repository during process, which should just run checks on it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,22 +19,28 @@
...
+    "terser:bench": "terser -c passes=2 -m --sequences 40 --module build/rollup/benchmark.umd.js -o build/benchmark-bundle.min.js",
+    "headers": "node ./tools/header.js build/peggy.min.js build/benchmark-bundle.min.js build/test-bundle.min.js",
+    "deploy": "npm run deploy:peggy && npm run deploy:tests && npm run deploy:bench",
+    "deploy:peggy": "cp build/peggy.min.js docs/vendor/peggy/peggy.min.js && cp build/peggy.min.js browser/peggy.min.js",
+    "deploy:tests": "cp build/test-bundle.min.js docs/js/test-bundle.min.js",
+    "deploy:bench": "cp build/benchmark-bundle.min.js docs/js/benchmark-bundle.min.js",
+    "build": "npm run build-notidy && npm run tidy",
     "build-notidy": "node ./tools/announce \"Static source analysis\" && npm run lint && node ./tools/announce \"Setting up the build\" && npm run setup && node ./tools/announce \"Making libraries\" && npm run make && node ./tools/announce \"Running Audits\" && npm run audit && node ./tools/announce \"Creating distributions\" && npm run headers && npm run deploy && node ./tools/announce \"Complete!\"",
```
Comment (hildjj) [inline package.json:39]: remove announce in this patch, please.

File: package.json [hunk 2]
```diff
@@ -19,22 +19,28 @@
...
-    "tidy": "rimraf -f build",
-    "build": "npm run build-notidy && npm run tidy"
+    "setup": "npm run clean && npm run set_version",
+    "make": "npm run parser && npm run ts && npm run rollup && npm run terser ",
```
Comment (hildjj) [inline package.json:41]: extra space at the end

File: package.json (hunk 1 above)
Comment (hildjj) [inline package.json:39]: should this be `build:notidy`?

**Review by hildjj — COMMENTED** [review]
works for me.  minor changes requested.

File: package.json (hunk 1 above)
Comment (Mingun) [reply package.json:39]: You mean, remove just calling `node ./tools/announce` or `build`, `build-notidy` and `tidy` completely?

Comment (hildjj) [reply package.json:39]: remove the script `tools/announce` and all of its use in `package.json`.  Particularly now that the steps are more easy to see in GHA, there's no need for it.

Comment (Mingun) [reply package.json]: Yes, good catch. Changed

Comment (Mingun) [reply package.json]: Script removed

**Review by hildjj — COMMENTED** [review]
I'll merge this if you're ok with my minor patch.

**Comment by Mingun** [comment]
I'm agree, merge it. Actually, I use only `parser`, `lint` and `test` in my everyday work.

**Pull request merged**